nexus/src/app/background/blueprint_load.rs

```diff
@@ -0,0 +1,159 @@
+                    // We failed to read the blueprint. There's nothing to do
+                    // but log an error. We'll retry when we're activated again.
+                    let message = format!("{:#}", error);
+                    warn!(
+                        &log,
+                        "failed to read target blueprint";
+                        "error" => &message
+                    );
+                    json!({
+                        "error": 
+                        format!("failed to read target blueprint: {message}")})
+                }
+                (None, Ok(None)) => {
+                    // We haven't found a blueprint yet. Do nothing.
+                    json!({})
+                }
+                (Some(old), Ok(None)) => {
+                    // We have transitioned from having a blueprint to not
+                    // having one. This should not happen.
+                    let message = format!(
+                        "target blueprint with id {} was removed. There is no \
+                        longer any target blueprint",
+                        old.id
+                    );
+                    error!(&log, "{}", message);
+                    json!({"error": message})
+                }
+                (None, Ok(Some((_, new_target)))) => {
+                    // We've found a new target blueprint for the first time.
+                    // Save it and notify any watchers.
+                    let target_id = new_target.id.to_string();
+                    let time_created = new_target.time_created.to_string();
+                    info!(
+                        log,
+                        "found new target blueprint (first find)";
+                        "target_id" => &target_id,
+                        "time_created" => &time_created
+                    );
+                    self.last = Some(Arc::new(new_target));
+                    self.tx.send_replace(self.last.clone());
+                    json!({
+                        "target_id": target_id, "time_created": time_created
+                    })
+                }
+                (Some(old), Ok(Some((_, new)))) => {
+                    let target_id = new.id.to_string();
+                    let time_created = new.time_created.to_string();
+                    if old.id != new.id {
+                        // The current target blueprint has been updated
+                        info!(
+                            log,
+                            "found new target blueprint";
+                            "target_id" => &target_id,
+                            "time_created" => &time_created
+                        );
+                        self.last = Some(Arc::new(new));
+                        self.tx.send_replace(self.last.clone());
+                        json!({
+                            "target_id": target_id,
+                            "time_created": time_created
+                        })
```

Comment:
Also: it'd be great to add code to `omdb` to parse this: a mention [here](https://github.com/oxidecomputer/omicron/blob/45df2e6ce738eba9a66e2f885bdc509d8932834b/dev-tools/omdb/src/bin/omdb/nexus.rs#L258) but mostly [here](https://github.com/oxidecomputer/omicron/blob/45df2e6ce738eba9a66e2f885bdc509d8932834b/dev-tools/omdb/src/bin/omdb/nexus.rs#L327-L339).  This isn't a blocker because omdb will do its best to print what it finds anyway but it'd be nice to have a human-readable summary there.

Reply:
Hmm, I started to do this, but the fields in each json object aren't necessarily the same and the whole thing would be a bit of a mess. Let me think about it.

Reply:
Happy to defer to y'all on saga output, but would it make sense to have some kind of serializable status enum that both this crate and omdb could use? It would round-trip through `serde_json` but would address the immediate issue of all the different fields.

Reply:
Yeah, there's:

- put the actual type into the OpenAPI spec and use a progenitor-generated type (seems ideal, but this seems hard to do without making things a lot more complicated?)
- define a struct to use on both sides (seems fine?)
- duplicate the struct on both sides (this is what we do today for the most part, and it's obviously not great, but the existing structs are also pretty simple, and the cost of getting it wrong is low)

They seem to all have their tradeoffs and I don't have a strong feeling.